### PR TITLE
feat: render unknown supplied children to UI

### PR DESCRIPTION
**Pull request by dwelle**
fix https://github.com/excalidraw/excalidraw/issues/6089

### CODE REVIEW

File: src/components/LayerUI.tsx [hunk 1]
```diff
@@ -111,8 +111,8 @@ const LayerUI = ({
 }: LayerUIProps) => {
   const device = useDevice();
 
-  const childrenComponents =
-    ReactChildrenToObject<UIChildrenComponents>(children);
+  const [childrenComponents, restChildren] =
+    partitionKnownChildren<UIChildrenComponents>(children);
```
Comment (dwelle) [inline src/components/LayerUI.tsx:115]: not the best API and naming (used from lodash), but didn't come up with a better one

File: src/components/LayerUI.tsx [hunk 2]
```diff
@@ -111,8 +111,11 @@ const LayerUI = ({
 }: LayerUIProps) => {
   const device = useDevice();
 
-  const childrenComponents =
-    ReactChildrenToObject<UIChildrenComponents>(children);
+  const [childrenComponents, restChildren] =
+    partitionReactChildren<UIChildrenComponents>(children, {
```
Comment (ad1992) [inline src/components/LayerUI.tsx:115]: `splitReactChildren` ?

**Review by ad1992 — COMMENTED** [review]
With this host will be able to render anything in side Excalidraw which we have been avoiding with filtering. 
so with this change host will be able to render anything inside excalidraw (any top-level children), I am not clear why we would want to support that.

**Comment by dwelle** [comment]
> With this host will be able to render anything in side Excalidraw which we have been avoiding with filtering.

I didn't know we were avoiding this, why?

File: src/components/LayerUI.tsx (hunk 2 above)
Comment (dwelle) [reply src/components/LayerUI.tsx:115]: split is related to strings. But we can do just `getReactChildren`.

**Review by ad1992 — APPROVED** [review]

**Pull request merged**